drivers/ltc4150/ltc4150_saul.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ltc4150
+ * @{
+ *
+ * @file
+ * @brief       SAUL adaption for LTC4150 devices
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "phydat.h"
+#include "saul.h"
+#include "ltc4150.h"
+
+static int read_total(const void *_dev, phydat_t *res)
+{
+    ltc4150_dev_t *dev = (ltc4150_dev_t *)_dev;
+    int32_t temp[3];
+
+    if (ltc4150_total(dev, temp) == 0) {
+        res->scale = -3;
+        res->unit = UNIT_COULOMB;
+#ifdef MODULE_LTC4150_BATTERY
+        /* Changing scale until value fits int16_t */
+        while ((temp[1] > INT16_MAX) || (temp[2] > INT16_MAX)) {
```

Comment:
[phydat_fit](https://github.com/RIOT-OS/RIOT/blob/master/sys/phydat/phydat.c#L27) could be your friend ;)

Reply:
Indeed

Reply:
The API of `phydat_fit` is way more complex to use than needed. I think this would make more sense:

- `void phydat_fit(phydat_t *dat, long *values, unsigned dim)`

Or with only three dimensions maybe even

- `uint8_t phydat_fit1(phydat_t *dat, long value)`
- `uint8_t phydat_fit2(phydat_t *dat, long val1, long val2)`
- `uint8_t phydat_fit3(phydat_t *dat, long val1, long val2, long val3)`

I'll prepare a separate PR for that.

Reply:
Done. I'm waiting for [that PR](https://github.com/RIOT-OS/RIOT/pull/9702) to be evaluated. If it get's merged, I can rebase against the master and use the new phydat_fit api.